Office 365/o365/ingest/parser.yml

```
@@ -320,6 +318,21 @@ stages:
           office365.investigation.alert.correlation_key: "{{parse_data.ParsedData.CorrelationKey}}"
           office365.investigation.alert.category: "{{parse_data.ParsedData.Category}}"
           office365.investigation.alert.source_type: "{{parse_data.ParsedData.SourceAlertType}}"
+          event.start: "{{json_event.message.StartTimeUtc}}"
+          event.end: "{{json_event.message.EndTimeUtc}}"
           host.name: "{{parse_data.ParsedData.MachineName}}"
+          log.level: "{{parse_data.ParsedData.Severity}}"
+          rule.name: "{{parse_data.ParsedData.AlertDisplayName}}"
+
       - set:
           event.kind: '{% if parse_data.ParsedData.IsIncident %}{{"alert"}}{% else %}{{"event"}}{% endif %}'
+
+      - set:
+          event.url: '[{% for Property in parse_data.ParsedData.Entities %}{% for Url in Property.Urls %} "{{Url}}", {% endfor %}{% endfor %}]'
+          source.ip: '[{% for Property in  parse_data.ParsedData.Entities %} "{{Property.SenderIP}}", {% endfor %}]'
```

`source.ip` is not a list. I suggest creating a custom field.
```suggestion
          office365.investigation.email.sender.ip: '[{% for Property in  parse_data.ParsedData.Entities %} "{{Property.SenderIP}}", {% endfor %}]'
```